model.py

```diff
@@ -3151,6 +3115,29 @@ def with_no_genres(self, q):
         q = q.filter(WorkGenre.genre==None)
         return q
 
+    @classmethod
+    def from_identifiers(cls, _db, identifiers, base_query=None):
+        """Finds all of the works indicated by a list of identifiers"""
+
+        identifier_ids = [identifier.id for identifier in identifiers]
+        if not identifier_ids:
+            return None
+
+        if not base_query:
+            # A raw base query that makes no accommodations for works that are
+            # suppressed or otherwise undeliverable.
+            base_query = _db.query(Work).join(Work.license_pools).\
+                join(LicensePool.identifier)
+
+        equivalent_identifier = aliased(Identifier)
+        query = base_query.outerjoin(Identifier.equivalencies).\
+                outerjoin(equivalent_identifier, Equivalency.output).\
+                filter(or_(
+                    Identifier.id.in_(identifier_ids),
+                    equivalent_identifier.id.in_(identifier_ids)
+                ))
```

Comment:
I think you should put a minimum threshold on Equivalency.strength so that someone's 2% opinion that two books are the same doesn't trigger a recommendation. Right now this won't matter because ISBN equivalencies directly associated with a commercial ID (such as an Overdrive ID) are very strong, but once we start getting LicensePools sold by ISBN, bad equivalencies will become a much bigger risk.

I would set the threshold at maybe 0.6 or 0.7.